clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy

```diff
@@ -193,12 +193,16 @@ class GoogleLoadBalancerCachingAgent extends AbstractGoogleCachingAgent implemen
         loadBalancers << newLoadBalancer
 
         if (forwardingRule.target) {
-          def targetPoolName = Utils.getLocalName(forwardingRule.target)
-          def targetPoolsCallback = new TargetPoolCallback(googleLoadBalancer: newLoadBalancer,
-                                                           httpHealthChecksRequest: httpHealthChecksRequest,
-                                                           instanceHealthRequest: instanceHealthRequest)
+          def forwardingRuleTokens = forwardingRule.target.split("/")
 
-          compute.targetPools().get(project, region, targetPoolName).queue(targetPoolsRequest, targetPoolsCallback)
+          if (forwardingRuleTokens[forwardingRuleTokens.size() - 2] != "targetVpnGateways") {
```

Comment:
Sounds good, I didn't know `[]` used `getAt()`

On Mon, Aug 8, 2016 at 9:41 AM, Matt Duftler <notifications@github.com>
wrote:

> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy
> <https://github.com/spinnaker/clouddriver/pull/860#discussion_r73876411>:
>
> >
> > -          compute.targetPools().get(project, region, targetPoolName).queue(targetPoolsRequest, targetPoolsCallback)
> > +          if (forwardingRuleTokens[forwardingRuleTokens.size() - 2] != "targetVpnGateways") {
>
> The url pattern shouldn't change since it is observed by all of our
> platform resources.
> And the [] lookup uses getAt() under the covers. That method won't throw
> ArrayIndexOutOfBoundsExceptions. It will return null if you're off of the
> end of the collection, and the final token if you're into the negatives.
> So feels ok to me.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/clouddriver/pull/860/files/0fa87968d8f032c1a54731c8afa448b59dead846#r73876411>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEpivcqKNZjDUKWqZVt86cy6jWUJHPV6ks5qdzIHgaJpZM4Jeq1b>
> .
>